[Samay Sagar] iP

reddevil1313 opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ _SUPER FAST_ to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉 

And it is **FREE**!

Features:

  - [x] Managing Tasks
  - [ ] Managing Deadlines (coming soon)
  - [ ] Reminders (coming soon)

---
If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

reddevil1313 committed:
Level 1 Greeting and echo messages

reddevil1313 committed:
Level 2 Adding tasks to a list

reddevil1313 committed:
Level 3 Mark as Done

reddevil1313 committed:
Level 4 Add support for tracking three types of tasks

reddevil1313 committed:
A-TextUiTesting Test using the I/O redirection technique

reddevil1313 committed:
Level-5: Handling Errors Test. Teach Duke to deal with errors such as incorrect inputs entered by the user.

reddevil1313 committed:
Level-6: Delete. Add support for deleting tasks from the list.

yatbun commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+            "\t" + "How can I help you?\n" + line;
+    private static ArrayList<Task> tasklist = new ArrayList<>();
+
+    private static void fuwwyEcho(String echo) {
```
Method name can more accurately describe its purpose.

yatbun commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+        fuwwyEcho("Uwu! I've marked this task as done:\n\t" + t + "\n");
+    }
+
+    private static void eorD(String command, String task) throws DukeException {
```
Method name can once again be more accurately descriptive. camelCase capitalisation should also be followed.

yatbun commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,34 @@
+    }
+
+    public boolean isDone() {
+        return isDone;
+    }
+
+    public void setName(String desc) {
+        this.description = desc;
```
Perhaps standardising whether to use the `this` keyword will help to improve readability.

yatbun commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+        Scanner sc = new Scanner(System.in);
+        String command = sc.nextLine();
+
+        while (!command.equals("bye")) {
+            try {
+                if (command.equals("list")) {
+                    String output = "\n";
+                    if (tasklist.size() == 0) {
+                        fuwwyEcho("Owo the list is empty boouwu");
+                    } else {
+                        for (int i = 0; i < tasklist.size(); i++) {
+                            int n = i + 1;
+                            String list = "\t" + n + ". " + tasklist.get(i);
+                            output += list + "\n";
+                        }
+                        fuwwyEcho(output);
+                    }
```
Good to avoid deep indentations past 3 levels.

yatbun reviewed (commented):
Overall, generally well done. Some method names can be improved and the overall legibility of the code can be improved by reducing indentation levels.

seannzx commented on src/main/java/Duke.java:
```diff
@@ -28,10 +28,15 @@ private static void taskDone(int n) {
     }
 
-    private static void eorD(String command, String task) {
+    private static void eorD(String command, String task) throws DukeException {
```
As mentioned above, I do also agree that method names should be more appropriate to facilitate better readability/understanding of code for future peer work!

seannzx commented on src/main/java/Duke.java:
```diff
@@ -38,8 +44,12 @@ public static void main(String[] args) {
+                taskDone(Integer.parseInt(command.split(" ")[1]));
+                command = sc.nextLine();
+            }
+            else {
```
Incorrect form for if-else statement

reddevil1313 committed:
Level 7: Save the tasks in the hard disk automatically.

reddevil1313 committed:
Changing the whole structure for Level 8

reddevil1313 committed:
Level 8: Taught Duke how to understand dates and times.

reddevil1313 committed:
Level 8: Merge Conflicts

reddevil1313 committed:
Level 8: Testing and Output Files

reddevil1313 committed:
A-MoreOOP: Refactor the code to extract out closely related code as classes.

reddevil1313 committed:
A-Packages: Organize the classes into suitable java packages

reddevil1313 committed:
A-JUnit: Add JUnit tests to test the behavior of the code.

reddevil1313 committed:
A-Jar: Package the app as an executable JAR file so that it can be distributed easily.

reddevil1313 committed:
A-JavaDoc: Add JavaDoc comments to the code.

reddevil1313 committed:
A-CodingStandard: Tweak the code to comply with a given coding standard

reddevil1313 committed:
Level-9: Give users a way to find a task by searching for a keyword.

reddevil1313 committed:
Merge branch 'branch-A-CodingStandard'

reddevil1313 committed:
Merge branch 'branch-Level-9'

reddevil1313 committed:
Merge branch 'add-gradle-support'

reddevil1313 committed:
Merge branch 'add-gradle-support'

reddevil1313 committed:
Merge branch 'branch-A-Gradle'

reddevil1313 committed:
A-Checkstyle: Omit for checkstyle errors and add checkstyle plugin

reddevil1313 committed:
Checkstyle for Test files

reddevil1313 committed:
Level 10: GUI

reddevil1313 committed:
A-Varargs: Add varargs to ChatbotUI

reddevil1313 committed:
A-Assertions: Add assertions for Parser

Assertions need to be added to document important assumptions that should hold at various points in the code.

Let's add statements to check the length of the command before parsing it.

Doing such will make sure that the command sent is of the correct input requirement.

reddevil1313 committed:
A-CodeQuality: Improve code quality

reddevil1313 committed:
Merge pull request #2 from reddevil1313/branch-A-Assertions

A-Assertions: Add assertions for Parser

reddevil1313 committed:
Merge branch 'master' into branch-A-CodeQuality

reddevil1313 committed:
Merge pull request #3 from reddevil1313/branch-A-CodeQuality

A-CodeQuality: Improve code quality

reddevil1313 committed:
BCD-Extension: Friendlier Syntax

Changed the input commands to 't', 'd', 'e' instead of 'todo', 'deadline' and 'extension' for efficient input.

reddevil1313 committed:
Add Ui.png

reddevil1313 committed:
A-UserGuide: Add User Guide

reddevil1313 committed:
Set theme jekyll-theme-slate

reddevil1313 committed:
Set theme jekyll-theme-cayman

reddevil1313 committed:
Set theme jekyll-theme-cayman

reddevil1313 committed:
A-BetterGui: Gui Adjustments

Change avatar pics, add application title and Pokémon theme.

reddevil1313 committed:
Theme change

reddevil1313 committed:
Merge branch 'master' of https://github.com/reddevil1313/ip

reddevil1313 committed:
Final Changes before IP Submission:

-JavaDocs
-CodeQuality
-Reuse credit
-Saving and Loading Files
-Error Handling
-Greeting Message
